### ISSUE TITLE
Can not access FTP or SSH

### ISSUE THREAD

**Issue by vdhong** [TechnikEmpire/CitadelCore.Windows#7]
Hello,
I run the test program and it worked good, but I can not connect FTP or SSH when the program running (I can connect when I turn off the program).
Please help to check this problem.
Thank you!

**Comment by TechnikEmpire** [comment]
@vdhong You're running the stock example from the repository? Or did you modify it in some way?

**Comment by TechnikEmpire** [comment]
Also, sorry for the delayed response. When I clicked the link on the notification initially it said 404 so I thought the issue was deleted or something.

**Comment by vdhong** [comment]
@TechnikEmpire 
Yes I modified it, In your example, it check only Chrome, I want to check all applications but I don't block it. Now I allow 2 ports 21 and 22 in OnFirewallCheck function and SSH application run ok but ftp can not get file list, ftp application can only connect then time out when request file list.

**Comment by TechnikEmpire** [comment]
@vdhong Can you post the modified project somewhere where I could test it?

**Comment by TechnikEmpire** [comment]
Based on that code, you're telling the filtering engine to filter every single non-system process. You need to be explicit about this stuff.

Ever since the 2.x update, the API is now port-specific in order to let users filter HTTP and HTTPS on whatever ports they feel like. But remember, it only filters HTTP. If you filter a non-http packet flow, you're going to be sending that flow to the internal filtering proxy that doesn't understand any other protocol.

In other words, you're sending FTP and such to a proxy that doesn't speak FTP protocol.

Only filter applications and packet flows that you are sure are HTTP/S.

**Comment by vdhong** [comment]
So can your library filter only HTTP(S) and pass other?

**Comment by TechnikEmpire** [comment]
@vdhong It can only do deep content inspection filtering on HTTP/HTTPS. Everything else, you can only do port + application based "filtering".

So for example we do something like this:

 - Allow google chrome to connect to example.com
 - Check and see if the request has the word "bad" in it.
 - If so, drop the connection and send a custom HTML response that says "don't send the word bad!"

You can also do
 - Allow filezilla to make a TCP connection to example.com on port 5050
 - Do not allow filezilla to make a connection to any domain on port 6060
 - Only allow firefox to connect to port 80 on remote addresses.
 - Etc, etc.

So basically aside from deep inspection of HTTP/HTTPS streams, you can use the library to act like a basic application firewall. Note that this functionality is a new addition in the 2.x version and currently only supports TCP. I have no plans to add UDP or other protocols because I'm not sure how much I want to commit to the use-case of building custom firewalls.

**Comment by vdhong** [comment]
Thank you.

**Comment by TechnikEmpire** [comment]
You're welcome. If you have any further questions or issues, feel free to open tickets.
